compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/NemoRunner.java

```diff
@@ -79,7 +81,9 @@ public NemoPipelineResult run(final Pipeline pipeline) {
     final PipelineVisitor pipelineVisitor = new PipelineVisitor(pipeline, nemoPipelineOptions);
     pipeline.traverseTopologically(pipelineVisitor);
     final NemoPipelineResult nemoPipelineResult = new NemoPipelineResult();
-    JobLauncher.launchDAG(pipelineVisitor.getConvertedPipeline(), nemoPipelineOptions.getJobName());
+    CompletableFuture.runAsync(() ->
+      JobLauncher.launchDAG(pipelineVisitor.getConvertedPipeline(), nemoPipelineOptions.getJobName()))
+      .thenRun(nemoPipelineResult::triggerJobDone);
     return nemoPipelineResult;
```

Comment:
I believe now we're expecting `NemoPipelineResult#getState` to return a value that is not `PipelineResult#RUNNING`, if `NemoPipelineResult#waitUntilFinish` was invoked first. Can you add a unit test to check that?